### PR TITLE
Add initial implementation for SD via Consul.

### PR DESCRIPTION
**Pull request by fabxc**
There is actually a `testutil` package for Consul. Unlike the API package it drags a tail of other external dependencies along. Vendoring all those might be a bit much compared to those 250 LOC to be tested.

### CODE REVIEW

File: retrieval/discovery/consul.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	consul "github.com/hashicorp/consul/api"
+
+	clientmodel "github.com/prometheus/client_golang/model"
```
Comment (juliusv) [inline retrieval/discovery/consul.go]: This should go together with the `consul` import in a separate group above.

File: retrieval/discovery/consul.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	consulWatchTimeout  = 30 * time.Second
+	consulRetryInterval = 15 * time.Second
+
+	// ConsuleNodeLabel is the name for the label containing a targets node name.
+	ConsulNodeLabel = clientmodel.MetaLabelPrefix + "consul_node"
+	// ConsulTagsLabel is the name of the label containing the tags assigned to the target.
+	ConsulTagsLabel = clientmodel.MetaLabelPrefix + "consul_tags"
+	// ConsulServiceLabel is the name of the label containing the service name.
+	ConsulServiceLabel = clientmodel.MetaLabelPrefix + "consul_service"
+	// ConsulDCLabel is the name of the label containing the datecenter ID.
```
Comment (juliusv) [inline retrieval/discovery/consul.go]: datecenter -> datacenter

**Comment by fabxc** [comment]
Done. Added `services` field to the config that defines which services we watch.

I did not add the defaulting to all services for now. Should we allow regexing we'd end up with two ways to do the same thing. Let's get this out there and figure out what people actually need.

**Comment by juliusv** [comment]
:+1:

**Pull request merged**